usaspending_api/broker/management/commands/fabs_nightly_loader.py

```diff
@@ -198,20 +200,27 @@ def handle(self, *args, **options):
             submission_ids = get_new_submission_ids(last_load_date)
             logger.info("Processing data for FABS starting from %s" % last_load_date)
 
-        if is_incremental_load and not submission_ids:
-            logger.info("No new submissions. Exiting.")
-
-        else:
+        if not is_incremental_load or submission_ids:
             with timer("obtaining delete records", logger.info):
                 ids_to_delete = get_fabs_records_to_delete(submission_ids, afa_ids, start_datetime, end_datetime)
 
             with timer("retrieving/diff-ing FABS Data", logger.info):
                 ids_to_upsert = get_fabs_transaction_ids(submission_ids, afa_ids, start_datetime, end_datetime)
 
-            update_award_ids = delete_fabs_transactions(ids_to_delete, do_not_log_deletions)
-            upsert_fabs_transactions(ids_to_upsert, update_award_ids)
+            externally_updated_award_ids = delete_fabs_transactions(ids_to_delete, do_not_log_deletions)
+            
+            if ids_to_upsert or externally_updated_award_ids:
+                update_award_ids = copy(externally_updated_award_ids)
+
+                if ids_to_upsert:
+                    with timer("inserting new FABS data", logger.info):
+                        update_award_ids.extend(insert_all_new_fabs(ids_to_upsert))
+
+            upsert_transactions(update_award_ids, "assistance")
```

Comment:
I'm confused what your asking. The if statement at 212 exists to guard this function call; it will always be true, no?

Reply:
My apologies.  This is what I get for reviewing on vacation.  What I MEANT to ask was, what happens here if `if ids_to_upsert or externally_updated_award_ids:` evaluates to *FALSE*.

I get `NameError: name 'update_award_ids' is not defined`.

This is because `upsert_transactions(update_award_ids, "assistance")` is not inside a code block where `update_award_ids` is defined.

Reply:
Ahh, a missing tab. Good eye.